core/src/main/java/org/apache/iceberg/RowKeyParser.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.util.Iterator;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+public class RowKeyParser {
+  private static final String ROW_KEY_ID = "key-id";
+  private static final String FIELDS = "fields";
+  private static final String SOURCE_ID = "source-id";
+
+  private RowKeyParser() {
+  }
+
+  public static void toJson(RowKey rowKey, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeNumberField(ROW_KEY_ID, rowKey.keyId());
+    generator.writeFieldName(FIELDS);
+    toJsonFields(rowKey, generator);
+    generator.writeEndObject();
+  }
+
+  public static String toJson(RowKey rowKey) {
+    return toJson(rowKey, false);
+  }
+
+  public static String toJson(RowKey rowKey, boolean pretty) {
+    try {
+      StringWriter writer = new StringWriter();
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
+      if (pretty) {
+        generator.useDefaultPrettyPrinter();
+      }
+      toJson(rowKey, generator);
+      generator.flush();
+      return writer.toString();
+
+    } catch (IOException e) {
+      throw new RuntimeIOException(e);
```

Comment:
We iceberg prefer to use the `UncheckedIOException`  rather than the deprecated `RuntimeIOException`, right ?